### PR TITLE
Add Alembic for Database Version Control

### PR DESCRIPTION
**Pull request by MichaelDeng03**
- Adds Alembic for database version control/migrations
- Add documentation on how to use Alembic
- Create initial Alembic migration script

### CODE REVIEW

**Comment by kgang** [comment]
:adequately-juiced:

**Comment by MichaelDeng03** [comment]
What does this emoji mean

**Comment by kgang** [comment]
i like to think its self-evident

File: docs/developer-guide.md [hunk 1]
```diff
@@ -78,20 +78,45 @@ def delete_my_table_core():
...
+SQLAlchemy requires a database URL. It's constructed using `sqlalchemy.URL` using environment variables. By default, the constructed URL will be a SQLite database.
+For a dockerized Postgres database, place the following into your `.env` file.
 
 ```.env
 DB_DRIVER=postgresql+psycopg
 DB_USERNAME=local_user
 DB_PASSWORD=local_password
 DB_PORT=5432
-DB_HOST=localhost
+DB_HOST=localhost
```
Comment (kgang) [inline docs/developer-guide.md:91]:
```suggestion
DB_HOST=localhost
```

Comment (kgang) [reply docs/developer-guide.md:91]: Could you please add these details for staging to 1password?

File: docs/developer-guide.md [hunk 2]
```diff
@@ -78,20 +78,45 @@ def delete_my_table_core():
...
+### Alembic
+
+We use Alembic to manage database migrations and schema creation locally for postgres.
+
+Migration scripts are tracked with git, and can be used to recreate database schemas at a particular point in time. This can be especially useful for testing staging/prod migrations, because we can recreate their schemas locally.
+
+#### Usage
+
+SQLModel models are used to define migration scripts.
+Import all defined models in `migrations/env.py`, e.g. `from concrete.db.orm.models import *`.
```
Comment (kgang) [inline docs/developer-guide.md:109]: Very nicely done

**Review by kgang — APPROVED** [review]
Looks incredible!

**Pull request merged**